src/MergeContext.js

```diff
@@ -454,6 +454,214 @@ class BranchPosition
+        // The default author will be used in the future commit if the 'Authored-by' attribute is missing.
+        this._defaultAuthor = defaultAuthor;
+
+        this._checkRaw(this._title);
+
+        if (rawPr.body !== undefined && rawPr.body !== null) {
+            const prDescription = rawPr.body.replace(/\r+\n/g, '\n');
+            this._checkRaw(prDescription);
+            this._parse(prDescription);
+        }
+    }
+
+    // complete message for the future commit
+    whole() {
+        assert(this._author !== null && this._body !== null && this._trailer !== null);
+        let message = this._title;
+        if (this._body)
+            message += '\n\n' + this._body;
+        if (this._trailer)
+            message += '\n\n' + this._trailer;
+        return message;
+    }
+
+    // the future commit author in the {name, email, date} format
+    author()
+    {
+        if (!this._author)
+            return this._defaultAuthor;
+        return {name: this._author.name, email: this._author.email, date: this._defaultAuthor.date};
+    }
+
+    // returns the position of the first non-ASCII_printable character (or -1)
+    _invalidCharacterPosition(str) {
+        const prohibitedCharacters = /[^\u{20}-\u{7e}]/u;
+        const match = prohibitedCharacters.exec(str);
+        return match ? match.index : -1;
+    }
+
+    // removes leading empty lines and trims the end
+    _trim(str) {
+        // cannot just use trim() to preserve the first line indentation (if any).
+        return str.replace(/^\s*\n+/g, '').trimEnd();
+    }
+
+    // basic checks for an unparsed message
+    _checkRaw(message) {
+        // CRs in CRLF sequences already removed
+        // other CRs are not treated specially (and are banned)
+        const trimmedMessage = this._trim(message);
+        const lines = trimmedMessage.split('\n');
+        for (let i = 0; i < lines.length; ++i) {
+            const line = lines[i];
+            const invalidPosition = this._invalidCharacterPosition(line);
+
+            if (invalidPosition !== -1)
+                throw new Error(`bad character at line ${i}, offset ${invalidPosition}`);
+
+            if (line.length > 72)
+                throw new Error(`too long line '${line}'`);
+        }
+    }
+
+    // parses attributes of the raw PR description
+    _parse(prDescription) {
+        const prDescriptionWithoutHeader = this._extractHeader(prDescription);
+
+        try {
+            const prDescriptionWithoutHeaderAndTrailer = this._extractTrailer(prDescriptionWithoutHeader);
+            this._parseBody(prDescriptionWithoutHeaderAndTrailer);
+        } catch (e) {
+            // TODO: supply debugString() prefix
+            Log.Logger.info("assuming no trailer: " + e.message);
+            this._parseBody(prDescriptionWithoutHeader);
+        }
+
+        assert(this._body !== undefined && this._body !== null);
+    }
+
+    // author is a {name, value}
+    _parseAuthor(author) {
```

Comment:
```suggestion
    // extracts author name and email from Authored-by and similar fields
    // returns a {name, email} author tuple
    _parseAuthor(author) {
        // author is a {name, value} tuple, where name is the field name
```

I also recommend renaming the author parameter to authorField. Otherwise, `author.name` expressions in this code confuse readers (like me) because an author also has a name.